include/tunnler/room_member.h

```diff
@@ -23,6 +24,14 @@
 // It also has to be used if you host a game yourself (You'd create both, a Room and a RoomMembership for yourself)
 class RoomMember final {
 public:
+    using Callback = std::shared_ptr<std::function<void()> >;
```

Comment:
is this common?

What kind of API did you look at to learn from? I feel we should replicate behaviour of successful C++ libs so our interface feels intuitive